src/main/java/duke/utils/Storage.java

```diff
@@ -0,0 +1,60 @@
+package duke.utils;
+
+import duke.DukeException;
+import duke.task.Task;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The `Storage` class handles reading and writing tasks to a data file.
+ */
+public class Storage {
+    private static final String DATA_FILE_PATH = "data.txt";
```

Comment:
I like how you use all uppercase and underscores for constant names, really improves readability 😄 .